packages/contracts/test/dollar/core/SecurityMonitorTest.sol

```diff
@@ -0,0 +1,138 @@
+    uint256 public constant INITIAL_LIQUIDITY = 1000;
+
+    function setUp() public {
+        vm.startPrank(admin);
+        mockUDollarToken = new MockUbiquityDollarToken();
+        mockUPool = new MockUbiquityPool();
+
+        mockUPool.setCollateralUsdBalance(INITIAL_LIQUIDITY);
+
+        securityMonitor = new SecurityMonitor(
+            address(mockUDollarToken),
+            address(mockUPool),
+            CHECK_INTERVAL,
+            INITIAL_LIQUIDITY
+        );
+        securityMonitor.grantRole(
+            securityMonitor.SECURITY_MONITOR_ROLE(),
+            user
+        );
+        vm.stopPrank();
+    }
+
+    function testInitialState() public {
+        assertEq(
+            address(securityMonitor.uDollarToken()),
+            address(mockUDollarToken)
+        );
+        assertEq(address(securityMonitor.uPool()), address(mockUPool));
+        assertTrue(
+            securityMonitor.hasRole(securityMonitor.DEFAULT_ADMIN_ROLE(), admin)
+        );
+        assertTrue(
+            securityMonitor.hasRole(
+                securityMonitor.SECURITY_MONITOR_ROLE(),
+                user
+            )
+        );
+    }
+
+    function testCheckUpkeep() public {
+        bool upkeepNeeded = securityMonitor.checkUpkeep("");
+        assertFalse(upkeepNeeded);
+
+        vm.warp(block.timestamp + CHECK_INTERVAL + 1);
+        upkeepNeeded = securityMonitor.checkUpkeep("");
+        assertTrue(upkeepNeeded);
+    }
+
+    function testPerformUpkeepUnauthorized() public {
+        vm.expectRevert("SecurityMonitor: not authorized");
+        securityMonitor.performUpkeep();
+    }
+
+    function testPerformUpkeepAuthorized() public {
+        vm.warp(block.timestamp + CHECK_INTERVAL + 1);
+
+        // Check that upkeep is needed before performing it
+        bool upkeepNeededBefore = securityMonitor.checkUpkeep("");
+        assertTrue(
+            upkeepNeededBefore,
+            "Upkeep should be needed before performUpkeep"
+        );
+
+        vm.prank(user);
+        securityMonitor.performUpkeep();
+
+        // Check that upkeep is not needed immediately after performing it
+        bool upkeepNeededAfter = securityMonitor.checkUpkeep("");
+        assertFalse(
+            upkeepNeededAfter,
+            "Upkeep should not be needed immediately after performUpkeep"
+        );
+    }
+
+    function testSecurityIncidentLiquidity() public {
+        mockUPool.setCollateralUsdBalance(50); // Set a low balance to trigger the incident
+
+        vm.warp(block.timestamp + CHECK_INTERVAL + 1);
+        vm.prank(user);
+
+        vm.expectEmit(true, false, false, true);
+        emit SecurityIncident("Liquidity below threshold");
+        securityMonitor.performUpkeep();
+
+        assertTrue(mockUDollarToken.paused(), "UDollarToken should be paused");
+        assertTrue(mockUPool.paused(), "UPool should be paused");
+    }
+
+    function testSecurityIncidentCollateralRatio() public {
```

Comment:
I don't understand the purpose of this test:
1. Collateral ration is set to 80%
2. Security monitor contract performs an upkeep
3. Suddenly everything is paused

What should we pause contracts in this case?